doc/dev-guide/reconciler.md

```diff
@@ -0,0 +1,59 @@
+# Operator design
+
+## Controllers and their managed resources
+
+### lvmcluster-controller
+
+- On receiving a valid LVMCluster CR, lvmcluster-controller reconciles the
+  following resource units for setting up [Topolvm](topolvm-repo) CSI and all
+  the supporting resources to use storage local to the node via Logical Volume
+  Manager (lvm)
+- *csiDriver*: Reconciles topolvm CSI Driver
+- *topolvmController*: Reconciles topolvm controller plugin
+- *lvmVG*: Reconciles volume groups from LVMCluster CR
+- *openshiftSccs*: Manages SCCs when the operator is run in Openshift
```

Comment:
The operator creates the SCCs? Security best practices is that the SCCs should be defined statically and not managed by an operator. For example, Rook defines its SCC in its upstream manifest [here](https://github.com/rook/rook/blob/master/deploy/examples/operator-openshift.yaml#L9). However, downstream doesn't have a way to define the SCCs in OLM, so the workaround was to have the operator do it. But upstream really should define the SCC in a standalone manifest so the operator doesn't need high privileges to create SCCs.

Reply:
- i may be wrong here, does rook being deployed standalone in downstream, isn't that setup by meta operators which also takes care of SCCs?
- if indeed rook upstream isn't setting SCCs from code, it does have to deploy necessary RBAC required to access SCC api irrespective of upstream/downstream, isn't it?

Reply:
> * i may be wrong here, does rook being deployed standalone in downstream, isn't that setup by meta operators which also takes care of SCCs?

Downstream Rook is only deployed as part of ODF and the OCS operator creates the necessary SCC since Rook does not create its own. 

> * if indeed rook upstream isn't setting SCCs from code, it does have to deploy necessary RBAC required to access SCC api irrespective of upstream/downstream, isn't it?

The Rook operator does not ever create SCCs or RBAC. It's all defined in its helm chart (for helm users), or the [common.yaml](https://github.com/rook/rook/blob/master/deploy/examples/common.yaml) for other upstream users. So the admin has full control to grant the operator only exactly the permissions needed.

Reply:
How should this be done for downstream outside of the operator? Bundles do not allow us to include SCCs so we decided to do this in the operator in order to make deployment easier.

Reply:
Downstream does require the operator to do it. I'm just suggesting that for upstream there should be an option for the operator not to do it, so the SCC can be created separately and avoid giving the operator the extra privileges to create SCCs.

Reply:
We can look into that.